cmd/notary/delegations.go

```diff
@@ -188,6 +193,11 @@ func (d *delegationCommander) delegationAdd(cmd *cobra.Command, args []string) e
 		}
 	}
 
+	// If the user passes --all-paths, give the "" path
+	if allPaths {
```

Comment:
I think this is similar to this: https://github.com/docker/notary/pull/514/files#r51305926

I think we should decide whether or not we should have any smart logic around adding/removing paths.  It would require changes to `AddPaths` and `RemovePaths` in `roles.go` -- right now they just merge/subtract string slices without any logic around the values of the included paths.